lego/apps/companies/views.py

```diff
@@ -54,10 +69,102 @@ class CompanyViewSet(
     ordering = "name"
 
     def get_serializer_class(self):
-        if self.action == "list":
-            return CompanyListSerializer
-
-        return CompanyDetailSerializer
+        return (
+            CompanyListSerializer if self.action == "list" else CompanyDetailSerializer
+        )
+
+    @action(detail=True, methods=["GET"])
+    def csv(self, *args, **kwargs):
+        user = self.request.user
+        is_admin = user.has_perm(EDIT, obj=Company)
+
+        if not is_admin:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        semester_year = kwargs["pk"].split("_")
```

Comment:
Could it be possible to format the received arguments in a more descriptive way? F.ex. changing the url from `companies/<pk>/csv` to `companies/csv/<year>/<month>/?optionalAttribute=lorem`

Right now the two mandatory params and the third optional params are all stored in the `pk` attribute, which makes it a bit hard to understand what it consists of and what types the values have